drivers/sensor/tmag5273/tmag5273.c

```diff
@@ -0,0 +1,1529 @@
+ * set the X/Y/Z angle & magnitude calculation mode
+ *
+ * @param dev handle to the sensor
+ * @param val value to be set
+ *
+ * @return \c -ENOTSUP if unknown value
+ * @return see @ref i2c_reg_update_byte_dt
+ */
+static inline int tmag5273_attr_set_xyz_calc(const struct device *dev,
+					const struct sensor_value *val)
+{
+	const struct tmag5273_config *drv_cfg = dev->config;
+	int retval = 0;
+
+	switch (val->val1) {
+		case TMAG5273_ANGLE_CALC_NONE: {
+			break;
+		}
+
+		case TMAG5273_ANGLE_CALC_XY: {
+			if (!drv_cfg->x_axis && !drv_cfg->y_axis) {
+				retval = -ENOTSUP;
+			}
+
+			break;
+		}
+
+		case TMAG5273_ANGLE_CALC_YZ: {
+			if (!drv_cfg->y_axis && !drv_cfg->z_axis) {
+				retval = -ENOTSUP;
+			}
+
+			break;
+		}
+
+		case TMAG5273_ANGLE_CALC_XZ: {
+			if (!drv_cfg->x_axis && !drv_cfg->z_axis) {
+				retval = -ENOTSUP;
+			}
+
+			break;
+		}
+
+		default: {
+			retval = -ENOTSUP;
+			break;
+		}
+	}
+
+	if (retval == 0) {
+		const struct tmag5273_config *drv_cfg = dev->config;
+
+		retval = i2c_reg_update_byte_dt(&drv_cfg->i2c,
+										TMAG5273_REG_SENSOR_CONFIG_2,
+										TMAG5273_ANGLE_EN_MSK,
+										val->val1 << TMAG5273_ANGLE_EN_POS);
+	} else {
+		LOG_ERR("unknown attribute value %d", val->val1);
+	}
+
+	return retval;
+}
+
+/**
+ * returns the X/Y/Z angle & magnitude calculation mode
+ *
+ * @param dev handle to the sensor
+ * @param val return value
+ *
+ * @return \c 0 on success
+ * @return see @ref i2c_reg_read_byte_dt
+ */
+static inline int tmag5273_attr_get_xyz_calc(const struct device *dev, struct sensor_value *val)
+{
+	const struct tmag5273_config *drv_cfg = dev->config;
+
+	uint8_t regdata = 0;
+	int retval = i2c_reg_read_byte_dt(&drv_cfg->i2c,
+										TMAG5273_REG_SENSOR_CONFIG_2,
+										&regdata);
+
+	if (retval != 0) {
+		return retval;
+	}
+
+	val->val1 = (regdata & TMAG5273_ANGLE_EN_MSK) >> TMAG5273_ANGLE_EN_POS;
+
+	return 0;
+}
+
+/* -------------------------------------------------------------------------------------- */
+
+static int tmag5273_attr_set(const struct device *dev,
+				enum sensor_channel chan,
+				enum sensor_attribute attr,
+				const struct sensor_value *val)
+{
+	if (dev == NULL) {
+		LOG_DBG("dev: NULL");
+		return -EINVAL;
+	}
+
+	if (val == NULL) {
+		LOG_DBG("val: NULL");
+		return -EINVAL;
+	}
+
+	int retval = 0;
+
+#ifdef CONFIG_PM_DEVICE
+	retval = tmag5273_pm_is_modifiable(dev);
+
+	if (retval != 0) {
+		return retval;
+	}
+
+#endif
+
+	if (chan != SENSOR_CHAN_MAGN_XYZ) {
+		return -ENOTSUP;
+	}
+
+	const struct tmag5273_config *drv_cfg = dev->config;
+
+	retval = -ENOTSUP;
+
+	switch ((uint16_t)attr) {
+	case TMAG5273_MEAS_RANGE: {
+		if (drv_cfg->meas_range == TMAG5273_AXIS_RANGE_RUNTIME) {
+			retval = tmag5273_attr_set_xyz_meas_range(dev, val);
+		}
+
+		break;
+	}
+	case TMAG5273_ATTR_CALC_AXIS: {
+		if (drv_cfg->angle_magnitude_axis == TMAG5273_ANGLE_MAG_RUNTIME) {
+			retval = tmag5273_attr_set_xyz_calc(dev, val);
+		}
+		break;
+	}
+	default: {
```

Comment:
This seems kind of weird. Wouldn't it make more sense to switch on the correct `enum` here?

I'm sort of questioning the value of moving all of the trigger code to a separate `.c` file.